### PR TITLE
feat: add `Dockerfile`

### PR DESCRIPTION
**Pull request by dni**
idea kilian and me discussed, can be useful in the future for people to run it themselves or for testing, etc.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM node
```
Comment (michael1011) [inline Dockerfile:1]: Set a tag for base images

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+WORKDIR /app
+COPY package.json .
```
Comment (michael1011) [inline Dockerfile:3]: The `package-lock.json` too

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+RUN npm i
```
Comment (michael1011) [inline Dockerfile:4]:
```suggestion
RUN npm ci
```

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+COPY . .
+
+ENV NETWORK=mainnet
+ENV NODE_ENV=docker
```
Comment (michael1011) [inline Dockerfile:8]: Not a valid value https://nodejs.org/en/learn/getting-started/nodejs-the-difference-between-development-and-production#nodejs-the-difference-between-development-and-production

File: docs/README.md [hunk 5]
```diff
@@ -12,6 +12,13 @@ coverY: 0
 
 Make sure to have the latest [Node.js LTS and NPM](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) installed. We recommend using [nvm](https://github.com/nvm-sh/nvm#install--update-script) to manage npm installs: `nvm install --lts`
 
+## with docker
```
Comment (michael1011) [inline docs/README.md:15]:
```suggestion
## With docker
```

File: vite.config.mjs [hunk 6]
```diff
@@ -4,6 +4,8 @@ import mkcert from "vite-plugin-mkcert";
 import { nodePolyfills } from "vite-plugin-node-polyfills";
 import solidPlugin from "vite-plugin-solid";
 
+const isNotDocker = process.env.NODE_ENV !== "docker";
```
Comment (michael1011) [inline vite.config.mjs:7]: Don't abuse `NODE_ENV` for this

File: vite.config.mjs [hunk 7]
```diff
@@ -12,7 +14,7 @@ const commitHash = child
 export default defineConfig({
     plugins: [solidPlugin(), mkcert(), nodePolyfills()],
     server: {
-        https: true,
+        https: isNotDocker,
```
Comment (michael1011) [inline vite.config.mjs:17]: Why no https via Docker?

**Review by michael1011 — CHANGES_REQUESTED** [review]

File: vite.config.mjs (hunk 7 above)
Comment (dni) [reply vite.config.mjs:17]: its better else you need to import the vite certs created inside the docker container or have cert errors. if you wont to expose it it better to run a proxy with ssl in front

Comment (michael1011) [reply vite.config.mjs:17]: We are working under the assumption that we are running with HTTPS. Eg clipboard API does not work when you are on HTTP. We need HTTPS

Comment (dni) [reply vite.config.mjs:17]: cclipboard works without htrtps, i think its just about the camera feature

Comment (michael1011) [reply vite.config.mjs:17]: > Only supported on pages served over HTTPS.
https://stackoverflow.com/a/30810322

HTTPS and localhost

**Review by michael1011 — APPROVED** [review]

**Pull request merged**